### PR TITLE
Switch to native compilation only for a more limited range of targets

### PR DESCRIPTION
**Pull request by RoastVeg**
~This still isn't functional due to platform differences on macos and windows.~

This PR add support for `--registry`, which overrides the default set in `Cargo.toml`, and adds binary releases for x86_64-unknown-linux-gnu and aarch64-apple darwin for `cargo-binstall`.

You can see how releases including binaries look here: https://github.com/SwitchEV/cargo-ktra-login/releases/tag/v0.1.4

### CODE REVIEW

**Review by Roger — APPROVED** [review]
looks good

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -34,73 +34,57 @@ jobs:
...
+        uses: actions/checkout@v4
 
-    - name: Install `cross` and compile artifact
-      run: |
-        cargo binstall \
-          --version=0.2.5 \
-          --no-confirm \
-          cross
+      - name: Fetch Rust toolchain
+        uses: dtolnay/rust-toolchain@1.74.0
```
Comment (OrangeTux) [inline .github/workflows/publish.yml:46]: What is the reasoning for installing cross via `dtolnay/rust-toolchain`?

**Review by OrangeTux — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (RoastVeg) [reply .github/workflows/publish.yml:46]: Cross is no longer used, only native targets are built. Using this GHA allows us to pin the rust version.

**Comment by OrangeTux** [comment]
`cargo-binstall` wasn't able to find the binaries for 0.1.4. See [here](https://github.com/SwitchEV/rocpp/actions/runs/9582390259/job/26421280780#step:5:13) for the logs of my build.

**Pull request merged**